<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>902</width>
    <height>689</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>902</width>
    <height>689</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(58, 9, 9);
color: rgb(255, 60, 60);
</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="styleSheet">
    <string notr="true">border-color: rgb(255, 20, 20);
border-width : 1.2px;
border-style: ridge;
margin-left: 4px;
margin-right: 4px;
padding:2px;
color: rgb(255, 76, 76);</string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <widget class="QLabel" name="label">
        <property name="maximumSize">
         <size>
          <width>121212</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Consolas</family>
          <pointsize>8</pointsize>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">color: rgb(253, 51, 51);</string>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;Enter the message you want to encode or decode.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="comboBox">
        <property name="minimumSize">
         <size>
          <width>431</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>200</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: rgb(0, 0, 0);</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLineEdit" name="inputStr">
        <property name="font">
         <font>
          <family>Consolas</family>
          <pointsize>10</pointsize>
          <weight>50</weight>
          <italic>false</italic>
          <bold>false</bold>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">
color: rgb(255, 58, 51);

background-color: rgb(20, 0, 0);</string>
        </property>
        <property name="maxLength">
         <number>1000</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="readFileButton">
        <property name="maximumSize">
         <size>
          <width>110</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">
background-color: rgb(0, 0, 0);
color: rgb(255, 110, 110);</string>
        </property>
        <property name="text">
         <string>Read fron file</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QPushButton" name="startCodingButton">
        <property name="maximumSize">
         <size>
          <width>150</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">
background-color: rgb(0, 0, 0);
color: rgb(255, 110, 110);</string>
        </property>
        <property name="text">
         <string>Launch</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="stepByStepStart">
        <property name="maximumSize">
         <size>
          <width>150</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">
background-color: rgb(0, 0, 0);
color: rgb(255, 110, 110);</string>
        </property>
        <property name="text">
         <string>Step-by-step launch</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QPushButton" name="readBit">
        <property name="maximumSize">
         <size>
          <width>150</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">
background-color: rgb(0, 0, 0);
color: rgb(255, 110, 110);</string>
        </property>
        <property name="text">
         <string>Read bit</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="addElemDecode">
        <property name="maximumSize">
         <size>
          <width>150</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: rgb(42, 0, 79);
color: rgb(255, 110, 110);</string>
        </property>
        <property name="text">
         <string>Add element</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="addElement">
        <property name="maximumSize">
         <size>
          <width>150</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">
background-color: rgb(0, 0, 0);
color: rgb(255, 110, 110);</string>
        </property>
        <property name="text">
         <string>Add element</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="upgradeTree">
        <property name="maximumSize">
         <size>
          <width>150</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: rgb(95, 0, 42);
color: rgb(255, 110, 110);</string>
        </property>
        <property name="text">
         <string>Update tree</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="prevStep">
        <property name="maximumSize">
         <size>
          <width>150</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">
background-color: rgb(0, 0, 0);
color: rgb(255, 110, 110);</string>
        </property>
        <property name="text">
         <string>Previous Step</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="stopButton">
        <property name="maximumSize">
         <size>
          <width>150</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">
background-color: rgb(0, 0, 0);
color: rgb(255, 110, 110);</string>
        </property>
        <property name="text">
         <string>Stop</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QLabel" name="label_2">
      <property name="maximumSize">
       <size>
        <width>121212</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="font">
       <font>
        <family>Consolas</family>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">color: rgb(253, 51, 51);</string>
      </property>
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;Binary tree view&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QGraphicsView" name="graphicsView">
      <property name="backgroundBrush">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>189</red>
         <green>4</green>
         <blue>4</blue>
        </color>
       </brush>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_5">
      <item>
       <widget class="QLabel" name="label_3">
        <property name="minimumSize">
         <size>
          <width>600</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>10000</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Consolas</family>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">color: rgb(253, 51, 51);</string>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;Resulting message&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_4">
        <property name="minimumSize">
         <size>
          <width>70</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>80</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Consolas</family>
         </font>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;right&quot;&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;Zoom:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="horizontalSlider">
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>200</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QTextBrowser" name="answer">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>40</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>230</height>
       </size>
      </property>
      <property name="font">
       <font>
        <family>Consolas</family>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">color: rgb(255, 58, 51);</string>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOn</enum>
      </property>
      <property name="sizeAdjustPolicy">
       <enum>QAbstractScrollArea::AdjustIgnored</enum>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="saveButton">
      <property name="styleSheet">
       <string notr="true">
background-color: rgb(0, 0, 0);
color: rgb(255, 110, 110);</string>
      </property>
      <property name="text">
       <string>Save results</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>902</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuInfo">
    <property name="title">
     <string>Info</string>
    </property>
    <addaction name="actionAbout_algorithm"/>
    <addaction name="actionInstruction"/>
   </widget>
   <addaction name="menuInfo"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionAbout_algorithm">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>About algorithm</string>
   </property>
  </action>
  <action name="actionInstruction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Instruction</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
